### PR TITLE
ACS-7473: [HxI LI Integration] Should return field values only as non-empty strings

### PR DESCRIPTION
**Pull request by krdabrowski**
[HxI LI Integration] Ingestion event handler should return field values only as non-empty strings
https://hyland.atlassian.net/browse/ACS-7473

### CODE REVIEW

File: live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/adapters/config/jackson/UpdateNodeEventSerializerTest.java [hunk 1]
```diff
@@ -92,43 +96,92 @@ public void shouldSerializePropertiesToSet()
...
     @Test
-    public void canCopeWithNullUsers()
+    public void shouldSerializeObjectPropertiesToSet()
     {
         UpdateNodeEvent event = new UpdateNodeEvent(NODE_ID, CREATE)
-                .addMetadataInstruction(new NodeProperty<>(CREATED_BY_PROPERTY, null))
-                .addMetadataInstruction(new NodeProperty<>(MODIFIED_BY_PROPERTY, null));
+                .addMetadataInstruction(new NodeProperty<>(CREATED_AT_PROPERTY, 10000L))
+                .addMetadataInstruction(new NodeProperty<>("someProperty", new Object() {
+                    @Getter
```
Comment (mpichura) [inline live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/adapters/config/jackson/UpdateNodeEventSerializerTest.java:150]: Nice trick with that `@Getter`

**Review by mpichura — APPROVED** [review]
LGTM - I had one idea regarding simplifying filtering properties traversal.

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/config/jackson/UpdateNodeEventSerializer.java [hunk 2]
```diff
@@ -96,6 +107,35 @@ public void serialize(UpdateNodeEvent event, JsonGenerator jgen, SerializerProvi
         }
     }
 
+    private NodeProperty<?> filterNonEmptyValueAndMap(NodeProperty<?> property)
```
Comment (mpichura) [inline live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/config/jackson/UpdateNodeEventSerializer.java:110]: Wouldn't it be neater if you returned `Optional`? That way you could get rid of those `null`s getting returned inside conditionals.

Comment (krdabrowski) [reply live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/config/jackson/UpdateNodeEventSerializer.java]: Good hint! Fixed

Comment (krdabrowski) [reply live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/config/jackson/UpdateNodeEventSerializer.java]: I went event further and finally even got rid of the `Optional`

**Pull request merged**